File: library/src/main/java/cz/ackee/rxoauth/AuthInterceptor.java
package cz.ackee.rxoauth;

import android.text.TextUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Interceptor for oauth2 authorization
 * Created by David Bilik[david.bilik@ackee.cz] on {03/03/16}
 **/
public class AuthInterceptor implements Interceptor {
    public static final String TAG = AuthInterceptor.class.getName();


    OAuthStore oAuthStore;

    public AuthInterceptor(OAuthStore oauthStore) {
        this.oAuthStore = oauthStore;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();
        Request.Builder builder = originalRequest.newBuilder();

        if (!TextUtils.isEmpty(oAuthStore.getAccessToken())) {
            String accToken = oAuthStore.getAccessToken();
            builder.addHeader("Authorization", "Bearer " + accToken);
        }
        return chain.proceed(builder.build());
    }
}

File: core/src/main/java/cz/ackee/rxoauth/AuthInterceptor.kt
package cz.ackee.rxoauth

import android.text.TextUtils
import okhttp3.Interceptor
import okhttp3.Response

/**
 * Interceptor that adds `Authorization: Bearer {access_token}` to all requests, if access_token is set.
 */
class AuthInterceptor(internal var oAuthStore: OAuthStore) : Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest = chain.request()
        val builder = originalRequest.newBuilder()

        if (!TextUtils.isEmpty(oAuthStore.accessToken)) {
            val accToken = oAuthStore.accessToken
            builder.addHeader("Authorization", "Bearer $accToken")
        }
        return chain.proceed(builder.build())
    }
}
